### PR TITLE
Utility for in-app URL's

### LINKED ISSUE #387: Utility for in-app URL's
**Is your feature request related to a problem? Please describe.**
Discord has a lot of protocols / uri's local in the client.
They might allow you to open the settings window, or open a specific user's profile.

Discord's buttons officially support these protocols, this means that you can make a button link to something (in-app)

To make these accessible to a normal programmer, a class should come with all URI's so everyone can implement one with ease.

**Describe the solution you'd like**
Utility class which generates URL's for the specific action you gave (based of an enum)

**Describe alternatives you've considered**
Hardcoded URL's, less accessible to the normal programmer. Most likely they won't know what exists and or are possible

**Additional context**
A link to all protocols can be found [here](https://gist.github.com/ghostrider-05/8f1a0bfc27c7c4509b4ea4e8ce718af0#file-discord_app_protocols-md) (not all work/more possibly exist)

- [x] general
- [x] settings
- [x] friends
- [x] user
- [x] guilds/dms
- [x] library
- [x] account

I've to take a look at:
- [ ] ~friends~
- [ ] ~guild-discovery~
- [x] routes with arguments

### PR DESCRIPTION
**Pull request by Tais993**
This implements a class which eases the developer to generate URL's to items within the Discord app.

A feature should **NOT** rely on this class, Discord can change something anytime, and iOS + Android aren't supported. They're there to enhance the functionality without making it worse.

If anyone has ideas for screens I could look at to find or there's URL(s) for, you can always send a message and I'll take a look or I can find anything.

Closes #387

### CODE REVIEW

**Review by Zabuzard — COMMENTED** [review]
Interesting. Can you give a few examples of how to use this class? Maybe also put them into the Javadoc, thanks.

**Comment by Tais993** [comment]
Added 2 short examples in the Javadoc, hopefully they explain enough?
@Zabuzard

**Review by Zabuzard — APPROVED** [review]
(fix the linter stuff though 👍 )

File: application/src/main/java/org/togetherjava/tjbot/commands/utils/DiscordClientAction.java [hunk 1]
```diff
@@ -107,12 +114,16 @@
     /**
      * Contains actions related to the settings menu
      */
-    public static class Settings {
+    @SuppressWarnings("squid:S2972")
```
Comment (Zabuzard) [inline application/src/main/java/org/togetherjava/tjbot/commands/utils/DiscordClientAction.java:117]: quick comment what it is and why it was disabled please

**Review by Zabuzard — CHANGES_REQUESTED** [review]

File: application/src/main/java/org/togetherjava/tjbot/commands/utils/DiscordClientAction.java (hunk 1 above)
Comment (Tais993) [reply application/src/main/java/org/togetherjava/tjbot/commands/utils/DiscordClientAction.java:122]: Done

Comment (Tais993) [reply application/src/main/java/org/togetherjava/tjbot/commands/utils/DiscordClientAction.java:122]: nvm, gonna rephrase

Comment (Zabuzard) [reply application/src/main/java/org/togetherjava/tjbot/commands/utils/DiscordClientAction.java:122]: Thanks. The linter is kinda right. The class isnt really designed nicely with all those constants and "abusing" the classes to enforce some sort of navigation/grouping between them... but I also dont really have any better suggestion right now, other than completely redesigning everything. Im okay with accepting this.

Comment (Tais993) [reply application/src/main/java/org/togetherjava/tjbot/commands/utils/DiscordClientAction.java:122]: Done v2

Comment (Tais993) [reply application/src/main/java/org/togetherjava/tjbot/commands/utils/DiscordClientAction.java:122]: > Thanks. The linter is kinda right. The class isnt really designed nicely with all those constants and "abusing" the classes to enforce some sort of navigation/grouping between them... but I also dont really have any better suggestion right now, other than completely redesigning everything. Im okay with accepting this.

What would your alternative design suggestion be then?

Comment (Zabuzard) [reply application/src/main/java/org/togetherjava/tjbot/commands/utils/DiscordClientAction.java:122]: I dont have anything concrete to be honest. I struggle finding anything better. I guess a "nice solution" would require to approach this with a completely different mindset.

**Review by Zabuzard — APPROVED** [review]

**Review by Heatmanofurioso — APPROVED** [review]

**Pull request merged**